taskfiles/utils.yml

```diff
@@ -141,3 +146,194 @@ tasks:
+        -B "{{.BUILD_DIR}}"
+        {{.CMAKE_ARGS}}
+      - >-
+        cmake
+        --build "{{.BUILD_DIR}}"
+        --parallel
+      # This command must be last
+      - task: "compute-checksum"
+        vars:
+          DATA_PATHS: ["{{.BUILD_DIR}}"]
+          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
+          EXCLUDE_PATHS: ["install_manifest.txt"]
+
+  # Runs cmake install step for the given build directory.
+  #
+  # @param {string} BUILD_DIR Cmake build directory.
+  # @param {string} INSTALL_PREFIX Path prefix for installing the project.
+  # @param {string={{.INSTALL_PREFIX}}.md5} [CHECKSUM_FILE] Path to store the checksum of installed
+  # files.
+  # @param {string[]=[{{.INSTALL_PREFIX}}]} [DATA_PATHS] Paths to compute the the checksum for.
+  cmake-install:
+    label: "{{.TASK}}-{{.BUILD_DIR}}-{{.INSTALL_PREFIX}}"
+    internal: true
+    vars:
+      CHECKSUM_FILE: >-
+        {{default (printf "%s.md5" .INSTALL_PREFIX) .CHECKSUM_FILE}}
+      # Convert the install prefix to a single element array in the default case where DATA_PATHS
+      # has not been set.
+      _INSTALL_PREFIX_ARRAY: ["{{.INSTALL_PREFIX}}"]
+      DATA_PATHS:
+        ref: "default ._INSTALL_PREFIX_ARRAY .DATA_PATHS"
+    requires:
+      vars: ["BUILD_DIR", "INSTALL_PREFIX"]
+    sources:
+      - "{{.BUILD_DIR}}/**/*"
+      - exclude: "{{.BUILD_DIR}}/install_manifest.txt"
+    generates: ["{{.CHECKSUM_FILE}}"]
+    deps:
+      - task: "validate-checksum"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          DATA_PATHS:
+            ref: ".DATA_PATHS"
+    cmds:
+      - >-
+        cmake
+        --install "{{.BUILD_DIR}}"
+        --prefix "{{.INSTALL_PREFIX}}"
+      # This command must be last
+      - task: "compute-checksum"
+        vars:
+          DATA_PATHS:
+            ref: ".DATA_PATHS"
+          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
+
+  # ===
+  # REMOTE UTILS
+  # ===
+
+  # Runs curl to download the provided URL.
+  #
+  # @param {string} URL
+  # @param {string} URL_SHA256 Content hash to verify downloaded file against.
+  # @param {string={{(base .URL)}}} [OUTPUT_FILE] File path to store the download file.
+  curl:
+    label: "{{.TASK}}-{{.OUTPUT_FILE}}"
+    internal: true
+    vars:
+      OUTPUT_FILE: "{{default (base .URL) .OUTPUT_FILE}}"
+    requires:
+      vars: ["URL", "URL_SHA256"]
+    generates:
+      - "{{.OUTPUT_FILE}}"
+    status:
+      - >-
+        diff
+        <(echo "{{.URL_SHA256}}")
+        <(openssl dgst -sha256 "{{.OUTPUT_FILE}}"
+        | awk '{print $2}')
+    cmds:
+      - |-
+        mkdir -p "{{dir .OUTPUT_FILE}}"
+        max_attempts=3
+        attempt=1
+        while [ $attempt -le $max_attempts ]; do
+          if curl --fail --location --show-error --connect-timeout 10 --max-time 300 \
+              --location "{{.URL}}" --output "{{.OUTPUT_FILE}}";
+          then
+            break
+          fi
+          echo "Attempt $attempt failed. Retrying..."
```

Comment:
Won't this print too many times? Another way to write it is to move the loop breaking condition down here. (Fwiw, I feel like I've seen this code before and commented on the same issue. I guess it's from a StackOverflow post, lol.)

Reply:
I'm just going to remove the log. The final log covers what happened in a failure anyway.

>Another way to write it is to move the loop breaking condition down here.
Fwiw, I'm pretty sure we'd need another condition to know whether we will retry again or not.

>(Fwiw, I feel like I've seen this code before and commented on the same issue. I guess it's from a StackOverflow post, lol.)
ya... it is from the rabbit